tests/test_putgb2.F90

```diff
@@ -22,79 +26,41 @@ program test_putgb2
   type(gribfield) :: gfld
   integer :: iret
 
-  jpdt(1) = 0
-  jpdt(2) = 0
-  jpdt(3) = 0
-  jpdt(4) = 0
-  jpdt(5) = 0
-  jpdt(6) = 12
-  jpdt(7) = 59
-  jpdt(8) = 0
-  jpdt(9) = 0
-  jpdt(10) = 1
-  jpdt(11) = 1
-  jpdt(12) = 1
-  jpdt(13) = 2
-  jpdt(14) = 1
-  jpdt(15) = 1
-
-  ! See https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_sect1.shtml.
-  jids(1) = 7 ! US National Weather Service - NCEP (WMC)
-  jids(2) = 4 ! Environmental Modeling Center
-  jids(3) = 2 ! GRIB master tables version number (currently 2)
-  jids(4) = 24 ! Version Implemented on 06 November 2019
-  jids(5) = 0 ! Local tables not used.
-  jids(6) = 2021 ! Year
-  jids(7) = 11 ! Month
-  jids(8) = 13 ! Day
-  jids(9) = 15 ! Hour
-  jids(10) = 59 ! Minute
-  jids(11) = 59 ! Second
-  jids(12) = 1 !  Operational Test Products
-  jids(13) = 0 ! Analysis Products
+  print *, 'Testing reading and writing of GRIB2 file.'
+  print *, 'testing reading...'
 
+  ! Open test file for reading.
   call baopenr(1, "testgrib.grb2", iret)
-  if (iret .ne. 0) then
-     print *, 'baopenr failed with iret value: ', iret
-     stop 3
-  end if
-
-  call getgb2(lugb, lugi, j, jdisc, jids, jpdtn, jpdt, jgdtn, jgdt, &
-       unpack, k, gfld, iret)
-  if (iret .ne. 0) then
-     print *, 'getgb2 failed with iret value: ', iret
-     stop 4
-  end if
-
-  call baclose(1, iret)
-  if (iret .ne. 0) then
-     print *, 'baclose failed with iret value: ', iret
-     stop 5
-  end if
+  if (iret .ne. 0) stop 3
 
+  ! Read a GRIB2 message.
+  call getgb2(lugb, lugi, j, jdisc, jids, jpdtn, jpdt, jgdtn, jgdt, unpack, k, gfld, iret)
+  if (iret .ne. 0) stop 4
+  if (gfld%ngrdpts .ne. 4) stop 5
   print *, 'gfld%ngrdpts from getgb2 is: ', gfld%ngrdpts
   print *, 'gfld%fld from getgb2 is: ', gfld%fld
 
-  print *, 'Testing open/write/close of GRIB2 file'
+  ! Close the file.
+  call baclose(1, iret)
+  if (iret .ne. 0) stop 20
+
+  print *, 'OK!'
+  print *, 'testing open/write/close of GRIB2 file...'
 
+  ! Open file for writing.
   call baopenw(2, "testgrib2.grb2", iret)
-  if (iret .ne. 0) then
-     print *, 'baopenr failed with iret value: ', iret
-     stop 6
-  end if
+  if (iret .ne. 0) stop 100
 
   call putgb2(2, gfld, iret)
-  if (iret .ne. 0) then
-     print *, 'putgb2 failed with iret value: ', iret
-     stop 7
-  end if
+  if (iret .ne. 0) stop 107
 
+  ! Close file.
   call baclose(2, iret)
-  if (iret .ne. 0) then
-     print *, 'baclose failed with iret value: ', iret
-     stop 8
-  end if
+  if (iret .ne. 0) stop 150
```

Comment:
For my own edification, what is the significance of the "stop" arguments? Are those error codes that are documented somewhere?

Reply:
No significance whatever. They are not error codes.

They could all be the same number in the test but then when it fails, you don't know where. So if you give each stop a different number, you can easily see where the failure is and fire up the debugger.

Test code is not user code, we don't have to give decent error messages or anything. (And that's why I took out these error messages in the test code.) The user should never see a test code failure, so it's only for us developers. Knowing where the stop happened is usually enough to know what is going on...